pkg/vm/vm.go

```diff
@@ -950,6 +955,36 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 
 		v.estack.PushItem(stackitem.NewBigInteger(new(big.Int).Sqrt(a)))
 
+	case opcode.MODMUL:
+		modules := v.estack.Pop().BigInt()
+		if modules.Sign() == 0 {
+			panic("zero module")
+		}
+		x2 := v.estack.Pop().BigInt()
+		x1 := v.estack.Pop().BigInt()
+
+		v.estack.PushItem(stackitem.NewBigInteger((new(big.Int)).Mod(x1.Mul(x1, x2), modules)))
```

Comment:
I don't think, it is safe to reuse `x1` here, we need to use newly allocated item for `x1` (also reuse it for `Mod` result).

Reply:
You're rignt, fixed.